Use assigneeId instead of their key or emails

Chemaclass opened the pull request:
# Description

Instead of passing around the raw array with the `jiraUsersToIgnore` (coming from the `.env`), I decided to "decorate" this and the logic behind (responsible to use that data) using the strategy pattern.

* I created a `StrategyFilter` interface and a concrete implementation named `NotWithAssigneeKeysFilter`.
* `TicketFilter` contains a factory method inside `notWithAssigneeKeys(string...)`. This is the class that actually knows how to use the `StrategyFilter`. 
* I moved all logic from `StrategyFilter` inside the `TicketsByAssignee` module.
* I moved all logic from `TicketsByAssignee` inside the `Jira` module.

Apart from that, I include a major refactor where it's being used the `assigneeId` instead of the assignee.key or assignee.email: both (key and emails) were removed from the JIRA Response (I guess due to some GDPR rules)

Chemaclass committed:
Creating IgnoreUsersPolicy

Chemaclass committed:
Create IgnoreStrategy::byAssigneeKey()

Chemaclass committed:
Rename it to TicketIgnorer

Chemaclass committed:
Rename from "Ignore" to "Filter"

Chemaclass committed:
Rename FilterByAssigneeKey to FilterNotWithAssigneeKeys

Chemaclass committed:
rename to StrategyFilter

JesusValeraDev reviewed (approved):
I don't understand pretty well the idea of the "StrategyFilter".
I mean, there is only one implementation, I am not sure if it worths to create the interface per se.

Chemaclass committed:
Add assigneeId and removed its email and key.
Because the JIRA response (I think due some GDPR issues) removed the assignee.email from their response when its not yourself

Chemaclass committed:
Update the .env.dist

Chemaclass committed:
Update README
